agent/bootstrap.go

```diff
@@ -864,7 +882,15 @@ func (b *Bootstrap) Start() error {
 			// gracefully handle repository renames
 			b.runCommand("git", "remote", "set-url", "origin", b.Repository)
 		} else {
-			b.runCommand("git", "clone", "-v", "--", b.Repository, ".")
+			// `git clone` can't accept options within quote like
+			// `git clone "-v --depth 1", so we need to split them
+			// by spaces and pass them individually.
+			gitCloneFlags := strings.Split(b.GitCloneFlags, " ")
+			gitCloneArguments := []string{"clone"}
+			gitCloneArguments = append(gitCloneArguments, gitCloneFlags...)
+			gitCloneArguments = append(gitCloneArguments, "--", b.Repository, ".")
```

Comment:
Here's a handy port of python's shlex: https://github.com/flynn-archive/go-shlex